test/common/alerts_firing.go

```diff
@@ -61,10 +76,99 @@ func (e *alertsFiringError) isValid() bool {
 	return !e.deadMansSwitchFiring || len(e.alertsFiring) != 0 || len(e.alertsPending) != 0
 }
 
-// TestIntegreatlyAlertsFiring reports any firing or pending alerts
 func TestIntegreatlyAlertsFiring(t *testing.T, ctx *TestingContext) {
 	var lastError error
 
+	if newErr := getFiringAlerts(t, ctx); newErr != nil {
+		lastError = newErr
+		podLogs(t, ctx)
+		t.Fatal(lastError.Error())
+	}
+
```

Comment:
Since we don't need the poll function here anymore, might be cleaner to get rid of the redundant `lastError` variable: 
```suggestion
	if err := getFiringAlerts(t, ctx); err != nil {
		podLogs(t, ctx)
		t.Fatal(err)
	}
```